### ISSUE TITLE
Feature Request: edit video with generated subtitle

### ISSUE THREAD

**Issue by momobobe** [absadiki/subsai#21]
https://github.com/zcf0508/autocut-client/ (package for https://github.com/mli/autocut) does a job to extract and assemble video clips based on subtitle generated by whisper, which is very useful. However, unfortunately that project looks lacking sustainable development and emerging modules like fast-whisper are not in their vision. So I wonder if such a feature can be implemented here to help more people with this use case.

**Comment by abdeladim-s** [comment]
@momobobe Could you please clarify more ? I didn't quite understand!

**Comment by abdeladim-s** [comment]
Yeah I understand now, so basically you mean basic video editing features like: cut, merge etc. 
I think it can be added, but it won't be as powerful as existing video editing softwares, because of the limitations of the UI library. 
I will add this feature to the TO-DO list.
Thanks @momobobe.

**Comment by m13v** [comment]
we built a transcript-driven video editor that does this. the workflow:

1. generate subtitles with word-level timestamps (fast-whisper works great for this)
2. user marks which segments to keep/cut either by text selection or by specifying keywords to search for
3. generate ffmpeg concat commands from the selected timestamp ranges

the tricky part is making clean cuts. cutting exactly at subtitle boundaries gives you jarring jumps. what works better: extend each segment by ~200ms on both sides for breathing room, then apply a short crossfade between segments (ffmpeg's xfade filter, 0.1-0.2s). this makes the cuts feel intentional rather than choppy.

for the actual ffmpeg command generation, we use the concat demuxer approach rather than the filter_complex approach. write each segment as a separate file reference with inpoint/outpoint, then concat them. this is way faster than re-encoding the whole video through filter_complex and preserves quality.

one gotcha with fast-whisper specifically: the timestamps can drift by ~100-200ms over long videos. we re-align by matching transcript text against a forced alignment pass on just the audio. adds a few seconds of processing but makes cuts much more precise.

**Comment by m13v** [comment]
our transcript-driven video editor is open source: https://github.com/m13v/video-edit - handles the whisper transcription -> segment selection -> ffmpeg concat workflow. might be useful as a reference implementation for the edit-by-subtitle feature.
